ui/app/components/job-status/panel.hbs

```diff
@@ -1,53 +1,114 @@
-<div class="job-status-panel boxed-section" data-test-job-status-panel data-test-status-mode={{@statusMode}}>
+<div class="job-status-panel boxed-section {{if this.isActivelyDeploying "active-deployment is-info"}}" data-test-job-status-panel data-test-status-mode={{@statusMode}}>
   <div class="boxed-section-head">
-    <h2>Status</h2>
-    <div class="select-mode">
-      <button type="button"
-        data-test-status-mode-current
-        class="button is-small is-borderless {{if (eq @statusMode "current") "is-active"}}"
-        {{on "click" (action @setStatusMode "current")}}
+    {{#if this.isActivelyDeploying}}
+      <div class="boxed-section-row"
+      {{did-insert (action this.establishOldAllocBlockIDs)}}
       >
-        Current
-      </button>
-      <button type="button"
-        data-test-status-mode-historical
-        class="button is-small is-borderless {{if (eq @statusMode "historical") "is-active"}}"
-        {{on "click" (action @setStatusMode "historical")}}>
-        Historical
-      </button>
-    </div>
+        Deployment Status
+        <span class="badge is-white bumper-left" data-test-active-deployment-stat="id">{{@job.latestDeployment.shortId}}</span>
+        <div class="pull-right">
+          {{#if @job.latestDeployment.isRunning}}
+            <TwoStepButton
+              data-test-fail
+              @classes={{hash
+                idleButton="is-danger"
+                confirmationMessage="inherit-color"
+                confirmButton="is-danger"}}
+              @idleText="Fail Deployment"
+              @cancelText="Cancel"
+              @confirmText="Yes, Fail Deployment"
+              @confirmationMessage="Are you sure?"
+              @inlineText={{true}}
+              @awaitingConfirmation={{this.fail.isRunning}}
+              @disabled={{this.fail.isRunning}}
+              @onConfirm={{perform this.fail}} />
+          {{/if}}
+          {{#if @job.latestDeployment.requiresPromotion}}
+            <button
+              data-test-promote-canary
+              type="button"
+              class="button is-warning is-small {{if this.promote.isRunning "is-loading"}}"
+              disabled={{this.promote.isRunning}}
+              onclick={{perform this.promote}}>Promote Canary</button>
+          {{/if}}
+        </div>
```

Comment:
A non-trivial part of this is ported verbatim from the pre-existing latest-deployment component

Reply:
My feeling is that it would be a wasted effort; I think we'll be generally reconsidering the nature of these confirmation buttons altogether soon and I wouldn't want to make a component API change that only lasts a month or two before that happens.